utils/sample_qc.py

```diff
@@ -0,0 +1,487 @@
+    callrate_data = np.matrix(data[pc_scores_ann].tolist())
+    logger.info('Assigning platforms to {} samples.'.format(len(callrate_data)))
+
+    # Cluster data
+    clusterer = hdbscan.HDBSCAN(min_cluster_size=hdbscan_min_cluster_size, min_samples=hdbscan_min_samples)
+    cluster_labels = clusterer.fit_predict(callrate_data)
+    n_clusters = len(set(cluster_labels)) - (-1 in cluster_labels)  # NOTE: -1 is the label for noisy (un-classifiable) data points
+    logger.info('Found {} unique platforms during platform imputation.'.format(n_clusters))
+
+    data['qc_platform'] = cluster_labels
+    ht = hl.Table.from_pandas(data, key=[*platform_pca_scores_ht.key])
+    ht = ht.annotate(qc_platform='platform_'+hl.str(ht.qc_platform))
+    return ht
+
+# TODO: This should be reviewed / merged with work from Kristen
+def impute_sex(
+        x_mt: hl.MatrixTable,  # TODO: This feels somewhat unsatisfying to provide two MTs. Maybe just reapply the QC MT filters to both (minus callrate for Y)?
+        y_mt: hl.MatrixTable,
+        platform_ht: hl.Table,
+        male_min_f_stat: float,
+        female_max_f_stat: float,
+        min_male_y_sites_called: int,
+        max_y_female_call_rate: float = 0.15,
+        min_y_male_call_rate: float = 0.8
+) -> hl.Table:
+    """
+    Imputes sample sex based on X-chromosome heterozygosity and Y-callrate (if Y calls are present)
+
+    :param x_mt:
+    :param y_mt:
+    :param platform_ht:
+    :param male_min_f_stat:
+    :param female_max_f_stat:
+    :param min_male_y_sites_called:
+    :param max_y_female_call_rate:
+    :param min_y_male_call_rate:
+    :return:
+    """
+    logger.info("Imputing samples sex")
+
+    x = hl.filter_intervals(x_mt, [hl.parse_locus_interval('X')])
+    x_ht = hl.impute_sex(x.GT, aaf_threshold=0.05, female_threshold=female_max_f_stat, male_threshold=male_min_f_stat)
+    y = hl.filter_intervals(y_mt, [hl.parse_locus_interval('Y')])
+    y = y.filter_rows(y.locus.in_y_nonpar())
+    sex_ht = y.annotate_cols(
+        qc_platform=platform_ht[y.col_key].qc_platform,
+        is_female=x_ht[y.col_key].is_female,
+        y_call_rate=hl.agg.fraction(hl.is_defined(y.GT)),
+        n_y_sites_called=hl.agg.count_where(hl.is_defined(y.GT)),
+        **{f'x_{ann}': x_ht[y.col_key][ann] for ann in x_ht.row_value if ann != 'is_female'}
+    ).cols()
+
+    mean_male_y_sites_called = sex_ht.aggregate(hl.agg.filter(~sex_ht.is_female, hl.agg.group_by(sex_ht.qc_platform, hl.agg.mean(sex_ht.n_y_sites_called))))
+    y_call_rate_stats = sex_ht.aggregate(
+        hl.agg.filter(
+            hl.is_defined(sex_ht.is_female),
+            hl.agg.group_by(hl.tuple([sex_ht.qc_platform, sex_ht.is_female]),
+                            hl.agg.stats(sex_ht.y_call_rate)
+                            )
+        )
+    )
+
+    no_y_call_rate_platforms = set()
+    for platform, sites_called in mean_male_y_sites_called.items():
+        if sites_called < min_male_y_sites_called:
```

Comment:
I've used `500` for `min_male_y_sites_called` --  I picked this just based on a reasonable amount of sites to compute metrics on. But could probably be lower.
For `max_y_female_call_rate`  and `min_y_male_call_rate`, it's based on plotting -- so it may not transfer well. Ultimately, this is one of the functions that is likely going to change since @klaricch also has a version. Hopefully we can merge them to get a great sex imputation function!